frontend/src/app/offers/offer-join-form/offer-join-form.component.ts

```diff
@@ -0,0 +1,69 @@
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {HttpClient} from '@angular/common/http';
+import {Location} from '@angular/common'
+
+import {ActivatedRoute} from '@angular/router';
+import {AuthService} from '../../auth.service';
+import {OffersService} from 'app/homepage-offer/offers.service';
+import {User} from '../../user';
+import {UserService} from '../../user.service';
+
+@Component({
+  selector: 'volontulo-offer-join-form',
+  templateUrl: './offer-join-form.component.html'
+})
+
+export class OfferJoinFormComponent implements OnInit {
+  public error;
+  public joinForm: FormGroup = this.fb.group({
+    applicantEmail: [{value: '', disabled: true}],
+    applicantName: [{value: '', disabled: true}],
+    message: ['', [Validators.minLength(10), Validators.maxLength(2000)]],
+    phoneNo: [{value: '', disabled: true}],
+    honeyValue: [''],
+  });
+  public offerId: number;
+  public submitEnabled = false;
+  public communicate = 'Dodatkowe informacje dla organizatora (pole nieobowiązkowe)';
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private authService: AuthService,
+    private fb: FormBuilder,
+    private httpClient: HttpClient,
+    private location: Location,
+    private offersService: OffersService,
+    private userService: UserService,
+  ) {
+  }
+
+  ngOnInit() {
+    this.authService.user$
+      .subscribe(
+        (user: User) => {
+          this.joinForm.controls.applicantEmail.setValue(user.email);
+          this.joinForm.controls.applicantName.setValue(this.userService.getFullName(user));
+          this.joinForm.controls.phoneNo.setValue(user.phoneNo);
+        }
+      );
+
+    this.activatedRoute.params
+      .switchMap(params => this.offerId = params.offerId)
+      .subscribe()
+  }
+
+  onSubmit() {
+    if (this.joinForm.valid) {
+      this.submitEnabled = true;
+      delete this.joinForm.value.honeyValue;
```

Comment:
I kind of unreflexively applied the solution used in contact component. I guess the original idea was not to pass the honeyValue along with the rest of form values, but you're right it makes no sense in here since all is getting passed in the body is the message.